quinn-proto/src/connection.rs

```diff
@@ -915,6 +915,7 @@ where
                 });
                 // No need to hold on to the stop_reason since it's propagated above
                 stream.state = ResetSent { stop_reason: None };
+                self.streams.maybe_cleanup(stream_id)
```

Comment:
This call, and the similar one added below, isn't needed. Cleanup of reset streams occurs when the reset is acknowledged (see `on_packet_acked`).

Reply:
Fixed.